Description: Upstream changes introduced in version 0.4-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 radare2 (0.4-1) unstable; urgency=low
 .
   * Initial release (Closes: #573345)
    + build system patch to add version to the libs
    + disable asm.m68k, BSD 4 clause is GPL incompatible
 .
 The person named in the Author field signed this changelog entry.
Author: Sebastian Reichel <elektranox@gmail.com>
Bug-Debian: http://bugs.debian.org/573345

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- radare2-0.4.orig/plugins.def.cfg
+++ radare2-0.4/plugins.def.cfg
@@ -35,7 +35,6 @@ SHARED="
 asm.bf
 asm.csr
 asm.dummy
-asm.m68k
 asm.psosvm
 asm.sparc
 anal.dummy
--- radare2-0.4.orig/libr/Makefile
+++ radare2-0.4/libr/Makefile
@@ -35,6 +35,9 @@ ifeq ($(OSTYPE),darwin)
 	@echo Linkage of libr2.dylib is broken on OSX because of dupped .o in .objs extraction
 else
 	${CC} -Wl,-soname=libr2.${EXT_SO} ${LDFLAGS_LIB} ${PIC_CFLAGS} .objs/*.o -o libr2.${EXT_SO}
+ifeq ($(OSTYPE),linux)
+	ln -s libr2.${EXT_SO} libr2.so
+endif
 endif
 
 libr2.${EXT_AR}:
@@ -91,6 +94,7 @@ install: install-includes install-pkgcon
 	@for a in `find * | grep -e '\.${EXT_SO}$$' | grep -v lib/t | grep lib` ; do \
 	  echo "$$a ${PFX}/lib"; \
 	  ${INSTALL_DATA} $$a ${PFX}/lib ; \
+	  ln -s `basename $$a` ${PFX}/lib/`basename $$a .0.0.0` ; \
 	done
 	# object archives
 	@for a in `find * | grep -e '\.a$$'` ; do \
--- radare2-0.4.orig/libr/rules.mk
+++ radare2-0.4/libr/rules.mk
@@ -28,6 +28,7 @@ ${LIBSO}: ${OBJ}
 	  if [ $$do = 1 ]; then \
 	    echo "${CC_LIB} ${LIBNAME} ${OBJ} ${LDFLAGS} ${LINK}" ; \
 	    ${CC_LIB} ${LIBNAME} ${OBJ} ${LDFLAGS} ${LINK} ; \
+		ln -s ${LIBSO} ${LIB}.${EXT_SO_PLUGIN} ; \
 	    if [ -f "../stripsyms.sh" ]; then sh ../stripsyms.sh ${LIBSO} ${NAME} ; fi ; \
 	  break ; \
 	fi ; done
--- radare2-0.4.orig/libr/config.mk.tail
+++ radare2-0.4/libr/config.mk.tail
@@ -54,13 +54,15 @@ ifeq (${OSTYPE},windows)
 CFLAGS+=-D__WINDOWS__=1
 EXT_AR=lib
 EXT_SO=dll
+EXT_SO_PLUGIN=dll
 EXT_EXE=.exe
 TH_LIBS=
 endif
 ifeq (${OSTYPE},gnulinux)
 CFLAGS+=-D__UNIX__=1
 EXT_AR=a
-EXT_SO=so
+EXT_SO=so.0.0.0
+EXT_SO_PLUGIN=so
 EXT_EXE=
 TH_LIBS=-lpthread
 endif
@@ -68,6 +70,7 @@ ifeq (${OSTYPE},darwin)
 CFLAGS+=-D__UNIX__=1
 EXT_AR=a
 EXT_SO=dylib
+EXT_SO_PLUGIN=dylib
 EXT_EXE=
 TH_LIBS=-lpthread
 endif
--- radare2-0.4.orig/libr/cmd/p/dummy.mk
+++ radare2-0.4/libr/cmd/p/dummy.mk
@@ -1,7 +1,7 @@
 OBJ_DUMMY+=cmd_dummy.o
 
 STATIC_OBJ+=${OBJ_DUMMY}
-TARGET_DUMMY=cmd_dummy.${EXT_SO}
+TARGET_DUMMY=cmd_dummy.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_DUMMY}
 
--- radare2-0.4.orig/libr/bp/p/x86.mk
+++ radare2-0.4/libr/bp/p/x86.mk
@@ -1,7 +1,7 @@
 OBJ_X86=bp_x86.o
 
 STATIC_OBJ+=${OBJ_X86}
-TARGET_X86=bp_x86.${EXT_SO}
+TARGET_X86=bp_x86.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_X86}
 
--- radare2-0.4.orig/libr/bp/p/arm.mk
+++ radare2-0.4/libr/bp/p/arm.mk
@@ -1,7 +1,7 @@
 OBJ_ARM=bp_arm.o
 
 STATIC_OBJ+=${OBJ_ARM}
-TARGET_ARM=bp_arm.${EXT_SO}
+TARGET_ARM=bp_arm.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_ARM}
 
--- radare2-0.4.orig/libr/bin/p/elf64.mk
+++ radare2-0.4/libr/bin/p/elf64.mk
@@ -2,7 +2,7 @@ OBJ_ELF64=bin_elf64.o bin_meta_elf64.o b
 OBJ_ELF64+=../format/elf/elf64.o ../format/elf/elf64_write.o
 
 STATIC_OBJ+=${OBJ_ELF64}
-TARGET_ELF64=bin_elf64.${EXT_SO}
+TARGET_ELF64=bin_elf64.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_ELF64}
 
--- radare2-0.4.orig/libr/bin/p/java.mk
+++ radare2-0.4/libr/bin/p/java.mk
@@ -1,7 +1,7 @@
 OBJ_JAVA=bin_java.o ../format/java/java.o
 
 STATIC_OBJ+=${OBJ_JAVA}
-TARGET_JAVA=bin_java.${EXT_SO}
+TARGET_JAVA=bin_java.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_JAVA}
 
--- radare2-0.4.orig/libr/bin/p/pe.mk
+++ radare2-0.4/libr/bin/p/pe.mk
@@ -1,7 +1,7 @@
 OBJ_PE=bin_pe.o ../format/pe/pe.o
 
 STATIC_OBJ+=${OBJ_PE}
-TARGET_PE=bin_pe.${EXT_SO}
+TARGET_PE=bin_pe.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PE}
 
--- radare2-0.4.orig/libr/bin/p/dummy.mk
+++ radare2-0.4/libr/bin/p/dummy.mk
@@ -1,7 +1,7 @@
 OBJ_DUMMY=bin_dummy.o
 
 STATIC_OBJ+=${OBJ_DUMMY}
-TARGET_DUMMY=bin_dummy.${EXT_SO}
+TARGET_DUMMY=bin_dummy.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_DUMMY}
 
--- radare2-0.4.orig/libr/bin/p/mach064.mk
+++ radare2-0.4/libr/bin/p/mach064.mk
@@ -1,7 +1,7 @@
 OBJ_MACH064=bin_mach064.o ../format/mach0/mach064.o
 
 STATIC_OBJ+=${OBJ_MACH064}
-TARGET_MACH064=bin_mach064.${EXT_SO}
+TARGET_MACH064=bin_mach064.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_MACH064}
 
--- radare2-0.4.orig/libr/bin/p/pe64.mk
+++ radare2-0.4/libr/bin/p/pe64.mk
@@ -1,7 +1,7 @@
 OBJ_PE64=bin_pe64.o ../format/pe/pe64.o
 
 STATIC_OBJ+=${OBJ_PE64}
-TARGET_PE64=bin_pe64.${EXT_SO}
+TARGET_PE64=bin_pe64.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PE64}
 
--- radare2-0.4.orig/libr/asm/p/sparc.mk
+++ radare2-0.4/libr/asm/p/sparc.mk
@@ -4,7 +4,7 @@ OBJ_SPARC+=../arch/sparc/gnu/sparc-opc.o
 
 STATIC_OBJ+=${OBJ_SPARC}
 
-TARGET_SPARC=asm_sparc.${EXT_SO}
+TARGET_SPARC=asm_sparc.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_SPARC}
 
 ${TARGET_SPARC}: ${OBJ_SPARC}
--- radare2-0.4.orig/libr/asm/p/java.mk
+++ radare2-0.4/libr/asm/p/java.mk
@@ -2,10 +2,10 @@ OBJ_JAVA=asm_java.o
 OBJ_JAVA+=../arch/java/javasm/javasm.o
 
 STATIC_OBJ+=${OBJ_JAVA}
-TARGET_JAVA=asm_java.${EXT_SO}
+TARGET_JAVA=asm_java.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_JAVA}
 
 ${TARGET_JAVA}: ${OBJ_JAVA}
-	${CC} -shared ${CFLAGS} -o asm_java.${EXT_SO} ${OBJ_JAVA}
-	@#strip -s asm_x86.${EXT_SO}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_JAVA}
+	@#strip -s $@
--- radare2-0.4.orig/libr/asm/p/dummy.mk
+++ radare2-0.4/libr/asm/p/dummy.mk
@@ -1,8 +1,8 @@
 OBJ_DUMMY+=asm_dummy.o
 
-TARGET_DUMMY=asm_dummy.${EXT_SO}
+TARGET_DUMMY=asm_dummy.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_DUMMY}
 STATIC_OBJ+=${OBJ_DUMMY}
 
 ${TARGET_DUMMY}: ${OBJ_DUMMY}
-	${CC} -shared ${CFLAGS} -o asm_dummy.${EXT_SO} asm_dummy.o
+	${CC} -shared ${CFLAGS} -o $@ asm_dummy.o
--- radare2-0.4.orig/libr/asm/p/csr.mk
+++ radare2-0.4/libr/asm/p/csr.mk
@@ -2,9 +2,9 @@ OBJ_CSR=asm_csr.o
 #OBJ_CSR+=../arch/csr/dis.o
 
 STATIC_OBJ+=${OBJ_CSR}
-TARGET_CSR=asm_csr.${EXT_SO}
+TARGET_CSR=asm_csr.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_CSR}
 
 ${TARGET_CSR}: ${OBJ_CSR}
-	${CC} -shared ${CFLAGS} -o asm_csr.${EXT_SO} ${OBJ_CSR}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_CSR}
--- radare2-0.4.orig/libr/asm/p/psosvm.mk
+++ radare2-0.4/libr/asm/p/psosvm.mk
@@ -2,9 +2,9 @@ OBJ_PSOSVM=asm_psosvm.o
 OBJ_PSOSVM+=../arch/psosvm/vmas/vmas.o
 
 STATIC_OBJ+=${OBJ_PSOSVM}
-TARGET_PSOSVM=asm_psosvm.${EXT_SO}
+TARGET_PSOSVM=asm_psosvm.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PSOSVM}
 
 ${TARGET_PSOSVM}: ${OBJ_PSOSVM}
-	${CC} -shared ${CFLAGS} -o asm_psosvm.${EXT_SO} ${OBJ_PSOSVM}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_PSOSVM}
--- radare2-0.4.orig/libr/asm/p/x86_olly.mk
+++ radare2-0.4/libr/asm/p/x86_olly.mk
@@ -4,7 +4,7 @@ OBJ_X86_OLLY+=../arch/x86/ollyasm/asmser
 OBJ_X86_OLLY+=../arch/x86/ollyasm/assembl.o
 
 STATIC_OBJ+=${OBJ_X86_OLLY}
-TARGET_X86_OLLY=asm_x86_olly.${EXT_SO}
+TARGET_X86_OLLY=asm_x86_olly.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_X86_OLLY}
 
--- radare2-0.4.orig/libr/asm/p/ppc.mk
+++ radare2-0.4/libr/asm/p/ppc.mk
@@ -3,9 +3,9 @@ OBJ_PPC+=../arch/ppc/gnu/ppc-dis.o
 OBJ_PPC+=../arch/ppc/gnu/ppc-opc.o
 
 STATIC_OBJ+=${OBJ_PPC}
-TARGET_PPC=asm_ppc.${EXT_SO}
+TARGET_PPC=asm_ppc.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PPC}
 
 ${TARGET_PPC}: ${OBJ_PPC}
-	${CC} -shared ${CFLAGS} -o asm_ppc.${EXT_SO} ${OBJ_PPC}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_PPC}
--- radare2-0.4.orig/libr/asm/p/x86.mk
+++ radare2-0.4/libr/asm/p/x86.mk
@@ -10,7 +10,7 @@ OBJ_X86+=../arch/x86/udis86/syn-intel.o
 OBJ_X86+=../arch/x86/udis86/syn-att.o
 
 STATIC_OBJ+=${OBJ_X86}
-TARGET_X86=asm_x86.${EXT_SO}
+TARGET_X86=asm_x86.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_X86}
 
--- radare2-0.4.orig/libr/asm/p/x86_nasm.mk
+++ radare2-0.4/libr/asm/p/x86_nasm.mk
@@ -1,7 +1,7 @@
 OBJ_X86_NASM=asm_x86_nasm.o
 
 STATIC_OBJ+=${OBJ_X86_NASM}
-TARGET_X86_NASM=asm_x86_nasm.${EXT_SO}
+TARGET_X86_NASM=asm_x86_nasm.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_X86_NASM}
 
--- radare2-0.4.orig/libr/asm/p/avr.mk
+++ radare2-0.4/libr/asm/p/avr.mk
@@ -1,10 +1,10 @@
 OBJ_AVR=asm_avr.o
 
 STATIC_OBJ+=${OBJ_AVR}
-TARGET_AVR=asm_avr.${EXT_SO}
+TARGET_AVR=asm_avr.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_AVR}
 
 ${TARGET_AVR}: ${OBJ_AVR}
-	${CC} -shared -I../arch/avr ${CFLAGS} -o asm_avr.${EXT_SO} ${OBJ_AVR}
-	@#strip -s asm_x86.${EXT_SO}
+	${CC} -shared -I../arch/avr ${CFLAGS} -o $@ ${OBJ_AVR}
+	@#strip -s $@
--- radare2-0.4.orig/libr/asm/p/bf.mk
+++ radare2-0.4/libr/asm/p/bf.mk
@@ -1,9 +1,9 @@
 OBJ_BF=asm_bf.o
 
-TARGET_BF=asm_bf.${EXT_SO}
+TARGET_BF=asm_bf.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_BF}
 STATIC_OBJ+=${OBJ_BF}
 
 ${TARGET_BF}: ${OBJ_BF}
 	${CC} -shared ${CFLAGS} -o ${TARGET_BF} ${OBJ_BF}
-	@#strip -s asm_x86.${EXT_SO}
+	@#strip -s asm_x86.${EXT_SO_PLUGIN}
--- radare2-0.4.orig/libr/asm/p/arm.mk
+++ radare2-0.4/libr/asm/p/arm.mk
@@ -2,10 +2,10 @@ OBJ_ARM=asm_arm.o
 OBJ_ARM+=../arch/arm/gnu/arm-dis.o
 
 STATIC_OBJ+=${OBJ_ARM}
-TARGET_ARM=asm_arm.${EXT_SO}
+TARGET_ARM=asm_arm.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_ARM}
 
 ${TARGET_ARM}: ${OBJ_ARM}
-	${CC} -shared ${CFLAGS} -o asm_arm.${EXT_SO} ${OBJ_ARM}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_ARM}
 	@#strip -s asm_x86.${EXT_SO}
--- radare2-0.4.orig/libr/asm/p/mips.mk
+++ radare2-0.4/libr/asm/p/mips.mk
@@ -4,7 +4,7 @@ OBJ_MIPS+=../arch/mips/gnu/mips-dis.o
 OBJ_MIPS+=../arch/mips/gnu/mips16-opc.o
 OBJ_MIPS+=../arch/mips/gnu/mips-opc.o
 
-TARGET_MIPS=asm_mips.${EXT_SO}
+TARGET_MIPS=asm_mips.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_MIPS}
 STATIC_OBJ+=${OBJ_MIPS}
 
--- radare2-0.4.orig/libr/asm/p/m68k.mk
+++ radare2-0.4/libr/asm/p/m68k.mk
@@ -2,9 +2,9 @@ OBJ_M68K=asm_m68k.o
 OBJ_M68K+=../arch/m68k/m68k_disasm/m68k_disasm.o
 
 STATIC_OBJ+=${OBJ_M68K}
-TARGET_M68K=asm_m68k.${EXT_SO}
+TARGET_M68K=asm_m68k.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_M68K}
 
 ${TARGET_M68K}: ${OBJ_M68K}
-	${CC} -shared ${CFLAGS} -o asm_m68k.${EXT_SO} ${OBJ_M68K}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_M68K}
--- radare2-0.4.orig/libr/parse/p/x86_pseudo.mk
+++ radare2-0.4/libr/parse/p/x86_pseudo.mk
@@ -1,9 +1,9 @@
 OBJ_X86PSEUDO+=parse_x86_pseudo.o
 
-TARGET_X86PSEUDO=parse_x86_pseudo.${EXT_SO}
+TARGET_X86PSEUDO=parse_x86_pseudo.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_X86PSEUDO}
 STATIC_OBJ+=${OBJ_X86PSEUDO}
 
 ${TARGET_X86PSEUDO}: ${OBJ_X86PSEUDO}
 	${CC} -L../../util -lr_util -shared ${CFLAGS} -o ${TARGET_X86PSEUDO} ${OBJ_X86PSEUDO}
-	@#strip -s asm_dummy.${EXT_SO}
+	@#strip -s asm_dummy.${EXT_SO_PLUGIN}
--- radare2-0.4.orig/libr/parse/p/mreplace.mk
+++ radare2-0.4/libr/parse/p/mreplace.mk
@@ -1,9 +1,9 @@
 OBJ_MREPLACE+=parse_mreplace.o parse_mreplace/mreplace.o parse_mreplace/mmemory.o
 
-TARGET_MREPLACE=parse_mreplace.${EXT_SO}
+TARGET_MREPLACE=parse_mreplace.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_MREPLACE}
 STATIC_OBJ+=${OBJ_MREPLACE}
 
 ${TARGET_MREPLACE}: ${OBJ_MREPLACE}
 	${CC} -shared ${CFLAGS} -o ${TARGET_MREPLACE} ${OBJ_MREPLACE}
-	@#strip -s asm_dummy.${EXT_SO}
+	@#strip -s $@
--- radare2-0.4.orig/libr/parse/p/dummy.mk
+++ radare2-0.4/libr/parse/p/dummy.mk
@@ -1,9 +1,9 @@
 OBJ_DUMMY+=parse_dummy.o
 
-TARGET_DUMMY=parse_dummy.${EXT_SO}
+TARGET_DUMMY=parse_dummy.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_DUMMY}
 STATIC_OBJ+=${OBJ_DUMMY}
 
 ${TARGET_DUMMY}: ${OBJ_DUMMY}
 	${CC} -shared ${CFLAGS} -o ${TARGET_DUMMY} ${OBJ_DUMMY}
-	@#strip -s asm_dummy.${EXT_SO}
+	@#strip -s asm_dummy.${EXT_SO_PLUGIN}
--- radare2-0.4.orig/libr/io/p/debug.mk
+++ radare2-0.4/libr/io/p/debug.mk
@@ -1,7 +1,7 @@
 OBJ_IODBG=io_debug.o
 
 STATIC_OBJ+=${OBJ_IODBG}
-TARGET_IODBG=io_debug.${EXT_SO}
+TARGET_IODBG=io_debug.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_IODBG}
 
--- radare2-0.4.orig/libr/io/p/malloc.mk
+++ radare2-0.4/libr/io/p/malloc.mk
@@ -1,7 +1,7 @@
 OBJ_MALLOC=io_malloc.o
 
 STATIC_OBJ+=${OBJ_MALLOC}
-TARGET_MALLOC=io_malloc.${EXT_SO}
+TARGET_MALLOC=io_malloc.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_MALLOC}
 
 ${TARGET_MALLOC}: ${OBJ_MALLOC}
--- radare2-0.4.orig/libr/io/p/shm.mk
+++ radare2-0.4/libr/io/p/shm.mk
@@ -1,7 +1,7 @@
 OBJ_SHM=io_shm.o
 
 STATIC_OBJ+=${OBJ_SHM}
-TARGET_SHM=io_shm.${EXT_SO}
+TARGET_SHM=io_shm.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_SHM}
 
 ${TARGET_SHM}: ${OBJ_SHM}
--- radare2-0.4.orig/libr/io/p/mach.mk
+++ radare2-0.4/libr/io/p/mach.mk
@@ -1,7 +1,7 @@
 OBJ_MACH=io_mach.o
 
 STATIC_OBJ+=${OBJ_MACH}
-TARGET_MACH=io_mach.${EXT_SO}
+TARGET_MACH=io_mach.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_MACH}
 
 ${TARGET_MACH}: ${OBJ_MACH}
--- radare2-0.4.orig/libr/io/p/ptrace.mk
+++ radare2-0.4/libr/io/p/ptrace.mk
@@ -1,7 +1,7 @@
 OBJ_PTRACE=io_ptrace.o
 
 STATIC_OBJ+=${OBJ_PTRACE}
-TARGET_PTRACE=io_ptrace.${EXT_SO}
+TARGET_PTRACE=io_ptrace.${EXT_SO_PLUGIN}
 ALL_TARGETS+=${TARGET_PTRACE}
 
 ${TARGET_PTRACE}: ${OBJ_PTRACE}
--- radare2-0.4.orig/libr/debug/p/native.mk
+++ radare2-0.4/libr/debug/p/native.mk
@@ -1,7 +1,7 @@
 OBJ_PTRACE=debug_native.o
 
 STATIC_OBJ+=${OBJ_PTRACE}
-TARGET_PTRACE=debug_native.${EXT_SO}
+TARGET_PTRACE=debug_native.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PTRACE}
 
--- radare2-0.4.orig/libr/debug/p/gdb.mk
+++ radare2-0.4/libr/debug/p/gdb.mk
@@ -15,7 +15,7 @@ libgdbwrap/gdbwrapper.o:
 	${CC} -c ${CFLAGS} -o libgdbwrap/gdbwrapper.o libgdbwrap/gdbwrapper.c
 
 STATIC_OBJ+=${OBJ_GDB}
-TARGET_GDB=debug_gdb.${EXT_SO}
+TARGET_GDB=debug_gdb.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_GDB}
 
--- radare2-0.4.orig/libr/lang/p/Makefile
+++ radare2-0.4/libr/lang/p/Makefile
@@ -5,7 +5,7 @@ include ../../config.mk
 CFLAGS=-I../../include -Wall -DPREFIX=\"${PREFIX}\"
 BINDEPS=
 
-LANGS=lang_dummy.${EXT_SO} lang_python.${EXT_SO} lang_perl.${EXT_SO}
+LANGS=lang_dummy.${EXT_SO_PLUGIN} lang_python.${EXT_SO} lang_perl.${EXT_SO}
 #LANGS+=lang_ruby.so
 ifeq ($(HAVE_LIB_TCC),1)
 LANGS+=lang_tcc.${EXT_SO}
@@ -48,8 +48,8 @@ lang_perl.${EXT_SO}:
 		`perl -MExtUtils::Embed -e ccopts` \
 		`perl -MExtUtils::Embed -e ldopts` -lncurses
 
-lang_dummy.${EXT_SO}: dummy.o
-	${CC} ${CFLAGS} -shared -fPIC ${LDFLAGS_LIB} -o lang_dummy.${EXT_SO} dummy.c ${LDFLAGS_LINKPATH}.. -L..
+lang_dummy.${EXT_SO_PLUGIN}: dummy.o
+	${CC} ${CFLAGS} -shared -fPIC ${LDFLAGS_LIB} -o $@ dummy.c ${LDFLAGS_LINKPATH}.. -L..
 	@#strip -s lua.so
 
 clean:
--- radare2-0.4.orig/libr/anal/p/dummy.mk
+++ radare2-0.4/libr/anal/p/dummy.mk
@@ -1,7 +1,7 @@
 OBJ_DUMMY=anal_dummy.o
 
 STATIC_OBJ+=${OBJ_DUMMY}
-TARGET_DUMMY=anal_dummy.${EXT_SO}
+TARGET_DUMMY=anal_dummy.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_DUMMY}
 
--- radare2-0.4.orig/libr/anal/p/ppc.mk
+++ radare2-0.4/libr/anal/p/ppc.mk
@@ -1,10 +1,10 @@
 OBJ_PPC=anal_ppc.o
 
 STATIC_OBJ+=${OBJ_PPC}
-TARGET_PPC=anal_ppc.${EXT_SO}
+TARGET_PPC=anal_ppc.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_PPC}
 
 ${TARGET_PPC}: ${OBJ_PPC}
-	${CC} -shared ${CFLAGS} -o anal_ppc.${EXT_SO} ${OBJ_PPC}
-	@#strip -s anal_ppc.${EXT_SO}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_PPC}
+	@#strip -s $@
--- radare2-0.4.orig/libr/anal/p/x86.mk
+++ radare2-0.4/libr/anal/p/x86.mk
@@ -2,10 +2,10 @@ OBJ_X86=anal_x86.o
 OBJ_X86+=../arch/x86/dislen.o
 
 STATIC_OBJ+=${OBJ_X86}
-TARGET_X86=anal_x86.${EXT_SO}
+TARGET_X86=anal_x86.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_X86}
 
 ${TARGET_X86}: ${OBJ_X86}
-	${CC} -shared ${CFLAGS} -o anal_x86.${EXT_SO} ${OBJ_X86}
-	@#strip -s anal_x86.${EXT_SO}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_X86}
+	@#strip -s $@
--- radare2-0.4.orig/libr/anal/p/arm.mk
+++ radare2-0.4/libr/anal/p/arm.mk
@@ -1,10 +1,10 @@
 OBJ_ARM=anal_arm.o
 
 STATIC_OBJ+=${OBJ_ARM}
-TARGET_ARM=anal_arm.${EXT_SO}
+TARGET_ARM=anal_arm.${EXT_SO_PLUGIN}
 
 ALL_TARGETS+=${TARGET_ARM}
 
 ${TARGET_ARM}: ${OBJ_ARM}
-	${CC} -shared ${CFLAGS} -o anal_arm.${EXT_SO} ${OBJ_ARM}
-	@#strip -s anal_x86.${EXT_SO}
+	${CC} -shared ${CFLAGS} -o $@ ${OBJ_ARM}
+	@#strip -s $@
